build/azure-pipelines.yml

```
@@ -134,6 +154,19 @@ stages:
               buildnumber="$(Build.BuildNumber)_$(Build.BuildId)_$sha"
               echo "##vso[build.updatebuildnumber]$buildnumber"
             displayName: Update build number
+          - script: |
+              npm version $(determineMajorVersion.fullVersion) --allow-same-version --no-git-tag-version
+              npm pack
+              mkdir publish
+              mv umbraco-cms-backoffice-$(determineMajorVersion.fullVersion).tgz publish/umbraco-cms-backoffice-latest.tgz
+              mv .npmrc publish/
+            displayName: Set version on Backoffice npm package
+            workingDirectory: src/Umbraco.Web.UI.New.Client
+          - task: PublishPipelineArtifact@1
+            displayName: Publish Backoffice NPM Artifact
+            inputs:
+              targetPath: src/Umbraco.Web.UI.New.Client/publish
+              artifactName: backoffice_npm_pack
```

Looks like `npm pack` has a parameter to specify the output directory, removing the need to manually move files afterwards: https://docs.npmjs.com/cli/v7/commands/npm-pack#pack-destination.

```suggestion
          - script: |
              npm version $(determineMajorVersion.fullVersion) --allow-same-version --no-git-tag-version
              npm pack --pack-destination $(Build.ArtifactStagingDirectory)/npm
            displayName: Set version on Backoffice npm package and publish
            workingDirectory: src/Umbraco.Web.UI.New.Client
          - task: PublishPipelineArtifact@1
            displayName: Publish Backoffice NPM Artifact
            inputs:
              targetPath: $(Build.ArtifactStagingDirectory)/npm
              artifactName: npm
```

---

Hmm, I see. The moving of files came as a need after using npm pack, so I didn't notice that option. Thanks for that. I see you are no longer moving the `.npmrc` file, which we need to be able to publish. Also I very much on purpose wanted to avoid using the version name in the artifact, since we also need to find this artifact and use it in the release pipelines and I couldn't see a way out of storing the version number for such use cases. Very prone to failure if someone changes something in the build pipelines and forgets about the release pipelines (nightly release).

What do you think about that?